build/frameworkbarrier/Dockerfile

```diff
@@ -0,0 +1,34 @@
+# MIT License
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE
+
+FROM golang:alpine
+
+ENV PROJECT_DIR=${GOPATH}/src/github.com/microsoft/frameworkcontroller
+
+RUN apk update && apk add bash && mkdir -p ${PROJECT_DIR}
+COPY . ${PROJECT_DIR}
+WORKDIR ${PROJECT_DIR}
+
+RUN ./build/frameworkbarrier/go-build.sh
+WORKDIR ./dist/frameworkbarrier
+
+ENTRYPOINT ["./start.sh"]
```

Comment:
You can use [multi-stage docker file](https://docs.docker.com/develop/develop-images/multistage-build/), this can reduce result image size significantly. [job-exporter](https://github.com/Microsoft/pai/blob/master/src/job-exporter/build/job-exporter.dockerfile#L24) used it.

Reply:
I do not think you need to install bash in the image, if it's for debug, you can still install in runtime container. Also recommend [fromlatest](https://www.fromlatest.io/#/) about tips to reduce image size.

Reply:
./start.sh need it

---
In reply to: [235229361](https://github.com/Microsoft/frameworkcontroller/pull/2#discussion_r235229361) [](ancestors = 235229361)

Reply:
Ah, ok.